DataFormats/TrackerRecHit2D/interface/VectorHit2D.h

```diff
@@ -0,0 +1,32 @@
+#ifndef TrackerRecHit2D_VectorHit2D_h
+#define TrackerRecHit2D_VectorHit2D_h
+
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/CLHEP/interface/AlgebraicObjects.h"
+#include "DataFormats/GeometrySurface/interface/LocalError.h"
+
+class VectorHit2D {
+public:
+  VectorHit2D() : thePosition(), theDirection(), theCovMatrix(), theChi2(), theDimension(2) {}
+  VectorHit2D(const LocalPoint& pos, const LocalVector& dir, const AlgebraicSymMatrix22& covMatrix, const double& Chi2)
+      : thePosition(pos), theDirection(dir), theCovMatrix(covMatrix), theChi2(Chi2), theDimension(2){};
+  virtual ~VectorHit2D(){};
+
+  LocalPoint localPosition() const { return thePosition; }
+  LocalVector localDirection() const { return theDirection; }
+  LocalError localDirectionError() const {
```

Comment:
Honestly I have no idea. I have switched the first two to return by const ref, the second one doesn't seem to work that way because that would be returning the reference to a temporary object.